python/faery/file_encoder.py

```diff
@@ -210,10 +229,51 @@ def frames_to_files(
                 for part in path_pattern.parts
             )
         )
-        path.parent.mkdir(exist_ok=True, parents=True)
         frame.to_file(
             path=path,
             compression_level=compression_level,
             file_type=file_type,
         )
+        state_manager.commit(frame=frame)
         index += 1
+    state_manager.end()
+
+
+def frames_to_file(
+    stream: collections.abc.Iterable[frame_stream.Rgba8888Frame],
+    path: typing.Union[pathlib.Path, str],
+    dimensions: tuple[int, int],
+    frame_rate: float = 60.0,
+    crf: float = 15.0,
+    preset: enums.VideoFilePreset = "ultrafast",
+    tune: enums.VideoFileTune = "none",
+    profile: enums.VideoFileProfile = "baseline",
+    file_type: typing.Optional[enums.VideoFileType] = None,
+    on_progress: typing.Callable[
+        [frame_stream.Rgba8888OutputState], None
+    ] = lambda _: None,
+):
+    path = pathlib.Path(path)
+    path.parent.mkdir(exist_ok=True, parents=True)
+    preset = enums.validate_video_file_preset(preset)
+    tune = enums.validate_video_file_tune(tune)
+    profile = enums.validate_video_file_profile(profile)
+    if file_type is None:
+        file_type = enums.video_file_type_guess(path)
+    else:
+        file_type = enums.validate_video_file_type(file_type)
+    state_manager = frame_stream.StateManager(stream=stream, on_progress=on_progress)
```

Comment:
Isn't the state manager redundant here? Maybe we could consider a default argument

Reply:
I am not sure what you mean by redundant.

Reply:
You're passing in the `on_progress` as an empty lambda (`lambda _: None`). Doesn't that mean the state manager is redundant? Or that the state isn't being updated?

Reply:
Oh, I see! The state manager is indeed redundant if the user does not specify an `on_progress` function, but it is useful if they do. We could optimize out the state manager when `on_progress` does nothing (we would probably want to use `None` as a default to do that), but it (hopefully) has minimal impact on performance (I have not checked though).